clippy_lints/src/assigning_clones.rs

```diff
@@ -0,0 +1,312 @@
+            TargetTrait::ToOwned => "use `clone_into()`",
+        }
+    }
+
+    fn suggested_replacement(
+        &self,
+        cx: &LateContext<'tcx>,
+        lhs: &hir::Expr<'tcx>,
+        applicability: &mut Applicability,
+    ) -> String {
+        match self.target {
+            TargetTrait::Clone => {
+                match self.kind {
+                    CallKind::MethodCall { receiver } => {
+                        let receiver_sugg = if let ExprKind::Unary(hir::UnOp::Deref, ref_expr) = lhs.kind {
+                            // `*lhs = self_expr.clone();` -> `lhs.clone_from(self_expr)`
+                            Sugg::hir_with_applicability(cx, ref_expr, "_", applicability)
+                        } else {
+                            // `lhs = self_expr.clone();` -> `lhs.clone_from(self_expr)`
+                            Sugg::hir_with_applicability(cx, lhs, "_", applicability)
+                        }
+                        .maybe_par();
+
+                        // Determine whether we need to reference the argument to clone_from().
+                        let clone_receiver_type = cx.typeck_results().expr_ty(receiver);
+                        let clone_receiver_adj_type = cx.typeck_results().expr_ty_adjusted(receiver);
+                        let mut arg_sugg = Sugg::hir_with_applicability(cx, receiver, "_", applicability);
+                        if clone_receiver_type != clone_receiver_adj_type {
+                            // The receiver may have been a value type, so we need to add an `&` to
+                            // be sure the argument to clone_from will be a reference.
+                            arg_sugg = arg_sugg.addr();
+                        };
+
+                        format!("{receiver_sugg}.clone_from({arg_sugg})")
+                    },
+                    CallKind::FunctionCall { self_arg, .. } => {
+                        let self_sugg = if let ExprKind::Unary(hir::UnOp::Deref, ref_expr) = lhs.kind {
+                            // `*lhs = Clone::clone(self_expr);` -> `Clone::clone_from(lhs, self_expr)`
+                            Sugg::hir_with_applicability(cx, ref_expr, "_", applicability)
+                        } else {
+                            // `lhs = Clone::clone(self_expr);` -> `Clone::clone_from(&mut lhs, self_expr)`
+                            Sugg::hir_with_applicability(cx, lhs, "_", applicability).mut_addr()
+                        };
+                        // The RHS had to be exactly correct before the call, there is no auto-deref for function calls.
+                        let rhs_sugg = Sugg::hir_with_applicability(cx, self_arg, "_", applicability);
+
+                        // TODO: how to get rid of the full path? Modify the function call function's (q)path? Or
+                        // auto-import it the trait?
+                        format!("::std::clone::Clone::clone_from({self_sugg}, {rhs_sugg})")
```

Comment:
`Clone` [is in the prelude](https://doc.rust-lang.org/std/prelude/index.html#:~:text=std%3A%3Aclone%3A%3AClone%2C%20the%20ubiquitous%20trait%20that%20defines%20clone%2C%20the%20method%20for%20producing%20a%20copy%20of%20a%20value.), no need to import it (?) (^･o･^)ﾉ”

This also includes to `ToOwned`

Reply:
I wrote a lot of similar lints/quick fixes for the IntelliJ Rust plugin, and there I was used to doing this the hard/proper way 😅 Although it's rare, someone could define their own Clone trait locally, and then the generated code wouldn't work.

Is this considered to be overkill for Clippy? :)

Reply:
If someone writes their own `Clone` trait, that person would be quite a silly goose. And should probably be reported to local authorities under the "excessive sillyness".

(It's just an edge case that accounting for it would remove QOL for the rest of the userbase)
Also, if the user decides to define their own Clone, adding an import to std::clone::Clone would result in an error because of the collision, there's a compiler lint preventing that (or maybe the imported Clone being ignored, I'm not sure).

Reply:
Ok, that makes things way easier :)